src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -3,26 +3,44 @@ import getProp from 'lodash/get';
 import { generatePath, match as matchType, matchPath } from 'react-router-dom';
 import { Location } from 'history';
 import AnnotatorContext from './AnnotatorContext';
-import { Action, Annotator, AnnotationActionEvent, AnnotatorState, GetMatchPath, MatchParams, Status } from './types';
+import { Action, AnnotationActionEvent, Annotator, AnnotatorState, GetMatchPath, MatchParams, Status } from './types';
+import { ElementsError } from '../../../types';
 
 const ANNOTATIONS_PATH = '/:sidebar/annotations/:fileVersionId/:annotationId?';
+const ERROR_CODE_FETCH_ANNOTATIONS = 'fetch_annotations_error';
+const ORIGIN_ACTIVITY_SIDEBAR = 'activity_sidebar';
+const ORIGIN_CONTENT_PREVIEW = 'content_preview';
+
+export interface SerializedError {
+    name?: string;
+    message?: string;
+    stack?: string;
+    code?: string;
+}
+
 export interface WithAnnotationsProps {
     onAnnotator: (annotator: Annotator) => void;
+    onAnnotatorCreate: (annotator: Annotator) => void;
+    onError?: (error: ElementsError, code: string) => void;
     onPreviewDestroy: (shouldReset?: boolean) => void;
 }
 
 export interface ComponentWithAnnotations {
     emitActiveChangeEvent: (id: string | null) => void;
     emitRemoveEvent: (id: string) => void;
+    escalateAnnotationsFetchError: (error: ElementsError) => void;
     getAction: (eventData: AnnotationActionEvent) => Action;
     getAnnotationsPath: (fileVersionId?: string, annotationId?: string | null) => string;
     getMatchPath: GetMatchPath;
     handleActiveChange: (annotationId: string | null) => void;
     handleAnnotationChangeEvent: (id: string | null) => void;
     handleAnnotationCreate: (eventData: AnnotationActionEvent) => void;
+    handleAnnotationsFetchError: ({ error }: { error: SerializedError }) => void;
```

Comment:
Does this need to have `SerializedError` type? Would the existing `Error` type work?